front/src/applications/editor/components/InfraErrors/InfraErrorCorrectorModal.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+
+import { ConfirmModal, useModal } from 'common/BootstrapSNCF/ModalSNCF';
+import { Operation, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import Loader from 'common/Loader';
+import { saveOperations } from 'reducers/editor';
+import { getInfraID } from '../../../../reducers/osrdconf/selectors';
+import { OPERATION_TYPE } from 'applications/editor/tools/types';
+
+const InfraErrorCorrectorModal: FC = () => {
+  const { t } = useTranslation();
+  const dispatch = useDispatch();
+  const { closeModal } = useModal();
+
+  const infraId = useSelector(getInfraID);
+  const [isLoading, setIsLoading] = useState(true);
+  const [isSaving, setIsSaving] = useState(false);
+  const [infraAutoFixes, setInfraAutoFixes] = useState<Operation[] | undefined>(undefined);
+
+  const [getInfraAutoFixes] = osrdEditoastApi.endpoints.getInfraByInfraIdAutoFixes.useLazyQuery();
+
+  useEffect(() => {
+    getInfraAutoFixes({ infraId: infraId as number })
+      .unwrap()
+      .then((results) => {
+        setInfraAutoFixes(results);
+        setIsLoading(false);
+      });
+  }, []);
```

Comment:
I think you can handle the endpoint like this, it requires less state and should work the same (you can also remove the isLoadingn state).
```suggestion
  const { data: infraAutoFixes, isLoading } =
    osrdEditoastApi.endpoints.getInfraByInfraIdAutoFixes.useQuery(
      { infraId: infraId as number },
      { skip: !infraId }
    );
```

Reply:
Bloody good proposition, thanks for that !